### PR TITLE
feat: Add `SQLITE3_SUPPORT` option [7076]

### LINKED ISSUE #921: Feature Request: Possibility to compile without the Sqlite3 plugin
( This is not a bug report, so I'm not using the bug report template. ) 

In our project, we are using FastRTPS and another commercial library which internally contains the same `sqlite3.c` that is used in this project which obviously causes duplicate definition issues in case of linking both of these libraries to the same target. 
As a matter of fact and since `sqlite3` is only used in a plugin, would it be possible to add a `CMAKE` option (which is **enabled** by default) to be able to build `FastRTPS` without this plugin?

### PR DESCRIPTION
**Pull request by alireza-hosseini**
## What I did

- Added `SQLITE3_SUPPORT` cmake option to enable/disable compilation of the `sqlite3PersistenceService`and its required `sqlite3` implementations. 
  - This option is **enabled** by default.
-Added `HAVE_SQLITE3` preprocessor flag to `config.h.in` which will be used to enable/disable `sqlite3PersisitenceService` related implementations.
- Used above flag to disable tests that are using `sqlite3`.
- Resolves #921 

## Why

As explained in #921, in some projects the `sqlite3PersistenceService` needs to be disabled because it requires to link the `sqlite3` into the **fastrtps** library which might cause "multiple definitions" errors in case the user's library or executable already links `sqlite3`internally.

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1056/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/877/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1054/)
* Performance [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Performance%20Tests%20Localhost/349/)

**Comment by alireza-hosseini** [comment]
@richiware It seems the **Mac** job just aborted unexpectedly and the **Performance** job seems to have problem with cloning the branch from a fork repo, right?
Should I push my branch to this repo?

File: examples/CMakeLists.txt [hunk 1]
```diff
@@ -43,3 +42,7 @@ add_subdirectory(C++/DDS)
 if(SECURITY)
     add_subdirectory(C++/SecureHelloWorldExample)
 endif()
+
+if(HAVE_SQLITE3)
```
Comment (richiware) [inline examples/CMakeLists.txt]:
```suggestion
if(SQLITE3_SUPPORT)
```

File: test/unittest/rtps/persistence/CMakeLists.txt [hunk 2]
```diff
@@ -16,7 +16,7 @@ if(NOT ((MSVC OR MSVC_IDE) AND EPROSIMA_INSTALLER))
     include(${PROJECT_SOURCE_DIR}/cmake/common/gtest.cmake)
     check_gtest()
 
-    if(GTEST_FOUND)
+    if(GTEST_FOUND AND HAVE_SQLITE3)
```
Comment (richiware) [inline test/unittest/rtps/persistence/CMakeLists.txt]:
```suggestion
    if(GTEST_FOUND AND SQLITE3_SUPPORT)
```

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1063/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/883/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1061/)

**Comment by alireza-hosseini** [comment]
@richiware Thanks for your review, change requests are addressed.

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1071/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/888/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1069/)

**Comment by alireza-hosseini** [comment]
@richiware Do you have any release planned in the near future because we need the changes made in this pull request in our project?

**Comment by richiware** [comment]
Thanks for you contribution.

**Comment by richiware** [comment]
About the release schedule, we released 1.8.3 and 1.9.3 two weeks ago. We haven't decided on the next releases yet. I can give you concrete info about this right now. sorry

**Pull request merged**